yearn/outputs/postgres/utils.py

```diff
@@ -0,0 +1,70 @@
+from brownie import Contract, chain, ZERO_ADDRESS
+from pony.orm import ObjectNotFound, db_session, TransactionIntegrityError, raw_sql
+from yearn.cache import memory
+from yearn.entities import Address, Token, UserTx, db
+from yearn.multicall2 import fetch_multicall
+from yearn.utils import is_contract
+
+chainid = chain.id
+
+@db_session
+@memory.cache()
+def cache_address(address: str) -> Address:
+    try:
+        return Address[chainid,address]
+    except ObjectNotFound:
+        try:
+            return Address(chainid=chainid,address=address,is_contract=is_contract(address))
+        except TransactionIntegrityError: # This can happen if two threads try to insert at same time
```

Comment:
The issue was that with 2 or more threads, you can end up in the following scenario: 
Thread A queries, no object is present, ObjectNotFound throws
Thread B queries, no object is present, ObjectNotFound throws
Thread A completes its insert
Thread B attempts insert which fails due to IntegrityError

I ended up just removing parallelism from the transactions exporter so more resources could be dedicated to the other exporters. So if you agree, I'll just remove this snippet from the code

Reply:
I propose not to use threads when it's not necessary. Will make life + code easier :sweat_smile:
If you want to use them, use `SELECT FOR UPDATE` or similar like banteg proposed and linked. This will basically ensure that only one thread will be able to process a given item and you can get rid of the exception handling.

Reply:
parallelism removed, will refactor this section now

Reply:
done